src/anndata/_core/storage.py

```diff
@@ -72,12 +72,23 @@ def coerce_array(
     allow_df: bool = False,
     allow_array_like: bool = False,
 ):
+    try:
+        # Needs to be done here to prevent circular imports, and StorageType is immutable
+        from anndata.experimental.backed._xarray import Dataset2D
+    except ImportError:
+
+        class Dataset2D:
+            @staticmethod
+            def __repr__():
+                return "mock anndata.experimental.backed._xarray."
+
     """Coerce arrays stored in layers/X, and aligned arrays ({obs,var}{m,p})."""
     # If value is a scalar and we allow that, return it
     if allow_array_like and np.isscalar(value):
         return value
     # If value is one of the allowed types, return it
-    if isinstance(value, StorageType.classes()):
+
+    if isinstance(value, StorageType.classes()) or isinstance(value, Dataset2D):
```

Comment:
It seems like there have been a few iterations on this functionality and at one point, this was used in a way where doing so would have broken other things (also, I think, why `pd.DataFrame` is not in it as below).  But that has since changed.

Reply:
Ah, right, https://github.com/scverse/anndata/pull/1247/files#diff-3b4874efa9bc9a522fa0cc5a88c69e233d7b46f2993331162bab12c4a6371bdcR75-R77

Reply:
We should probably fix than in your other PR then before having this hack in here. (also the comment still mentions `StorageType`, not `ArrayDataStructureType`)